addons/account/models/chart_template.py

```diff
@@ -184,7 +189,25 @@ def try_loading(self, company=False):
         if not company.chart_template_id and not self.existing_accounting(company):
             for template in self:
                 template.with_context(default_company_id=company.id)._load(15.0, 15.0, company)
-
+            # Install the demo data when the first localization is instanciated on the company
+            if install_demo and self.env.ref('base.module_account').demo:
+                self.with_context(
+                    default_company_id=company.id,
+                    allowed_company_ids=[company.id],
+                )._create_demo_data()
+
+    def _create_demo_data(self):
+        try:
+            demo_data = self._get_demo_data()
+            for model, record in demo_data:
+                created = self.env[model]._load_records([{
+                    'xml_id': "account.%s" % xml_id if '.' not in xml_id else xml_id,
+                    'values': data,
+                } for xml_id, data in record])
```

Comment:
That's not really what `_load_records` is for, and the number of records is not really extreme, maybe create and assign xids normally?

Reply:
What do you mean?
I am creating or updating records of this model and I assign XMLIDs.
```python
    def _load_records(self, data_list, update=False):
        """ Create or update records of this model, and assign XMLIDs.

            :param data_list: list of dicts with keys `xml_id` (XMLID to
                assign), `noupdate` (flag on XMLID), `values` (field values)
            :param update: should be ``True`` when upgrading a module

            :return: the records corresponding to ``data_list``
        """
```

Reply:
`_load_records` was created for the purpose of the import: bulk-creation/update of records and bulk-creation of xids. It's not intended as a general-purpose convenience method.

Reply:
Demo data are loaded using this method when they are declared in xml. The only thing I changed was the data declaration; how is this different?

Reply:
> Demo data are loaded using this method when they are declared in xml.

And that's an implementation detail of the import.

> The only thing I changed was the data declaration

By making it not a data import.

> how is this different?

In that it's not an import anymore.

Reply:
Ok, so I'll copy paste the method `_load_records` and give it another name. Got it.

Reply:
https://github.com/odoo/odoo/commit/d5b687a48ffbd9fc952b4fef7089b8c83399e2dd

@rco-odoo didn't seem to be that reluctant when he wrote that in `addons/account/models/chart_template.py`

Reply:
@xmo-odoo what other method do you propose to use? I find this one perfectly fitted